### PR TITLE
Don't log sensitive information

### PR DESCRIPTION
**Pull request by reline**
Never log user data or PII (Personal Identification Information) since other applications can access the log.
Android logs are not stripped unless using ProGuard, which not all apps use.

### CODE REVIEW

**Review by ferdicus — APPROVED** [review]
Looks good, thanks for contribution. 

Two questions popped in my head:
* did you check how iOS handles this?
* would be cool to give people the option to dis-/en-able this if they want
  * with the default being to not log it

I'll approve this one, would be cool if you could find the time to check the other two 👍 

Thanks 🙇

**Comment by reline** [comment]
Good point, I'll check iOS today.

As to customizing the logs, I think it's more effort than it's worth personally, but a simple `BuildConfig.DEBUG` check could be used to always log in debug, and never log in prod.

I figured the logs were leftover from testing the library anyways.

**Comment by reline** [comment]
I did find several logs that may prove to be sensitive on iOS, seems they aren't stripped at all unless using macros.

```objc
MGLCustomHeaders.m

NSLog(@"Add custom header %@ %@", headerName, value);
...
NSLog(@"Remove custom header %@", header);
```

Not sure about this one, I don't see any usages anywhere and don't know what `location` might be since it's a string.
```objc
RCTMGLVectorSourceManager.m

RCT_EXPORT_METHOD(addEvent:(NSString *)name location:(NSString *)location)
{
    RCTLogInfo(@"Pretending to create an event %@ at %@", name, location);
}
```

I'll go ahead and remove the header logs from iOS in this PR if that's okay.

File: ios/RCTMGL/MGLCustomHeaders.m [hunk 1]
```diff
@@ -15,24 +15,24 @@ +(NSMutableURLRequest*) __swizzle_requestWithURL:(NSURL*)url {
         if([url.scheme isEqualToString:@"ws"]) {
             return [NSMutableURLRequest __swizzle_requestWithURL:url];
         }
-        
+
```
Comment (ferdicus) [inline ios/RCTMGL/MGLCustomHeaders.m:18]: thanks for those newlines :)

finally some rest for the eyes while reading

**Review by ferdicus — APPROVED** [review]

**Comment by ferdicus** [comment]
> I did find several logs that may prove to be sensitive on iOS, seems they aren't stripped at all unless using macros.
> 
> ```objc
> MGLCustomHeaders.m
> 
> NSLog(@"Add custom header %@ %@", headerName, value);
> ...
> NSLog(@"Remove custom header %@", header);
> ```

👍 

> Not sure about this one, I don't see any usages anywhere and don't know what `location` might be since it's a string.
> 
> ```objc
> RCTMGLVectorSourceManager.m
> 
> RCT_EXPORT_METHOD(addEvent:(NSString *)name location:(NSString *)location)
> {
>     RCTLogInfo(@"Pretending to create an event %@ at %@", name, location);
> }
> ```

Hmm, yeah, hard to say... `Pretending to create an event` reads funny as well... not sure - lets let it in for now, maybe it's useful to someone.


> I'll go ahead and remove the header logs from iOS in this PR if that's okay.

Sure thing - that's fine

**Pull request merged**